tests/components/telegram_bot/test_telegram_bot.py

```diff
@@ -45,6 +46,27 @@ async def test_send_message(hass: HomeAssistant, webhook_platform) -> None:
     assert events[0].context == context
 
 
+async def test_send_message_thread(hass: HomeAssistant, webhook_platform) -> None:
+    """Test the send_message service for threads."""
+    context = Context()
+    events = async_capture_events(hass, "telegram_sent")
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SEND_MESSAGE,
+        {
+            ATTR_MESSAGE: "test_message",
+            ATTR_MESSAGE_THREAD_ID: "123"
+        },
+        blocking=True,
+        context=context,
+    )
+    await hass.async_block_till_done()
+
+    assert len(events) == 1
+    assert events[0].context == context
```

Comment:
This doesn't show that the thread id is passed to the API.
The test already uses the fixture `mock_external_calls` indirectly via the `webhook_platform` fixture. I suggest to modify the `mock_external_calls` fixture to yield the patches instead of yielding nothing, then you can test here that `telegram.Bot.send_message` was called once with the correct data.

Reply:
Hi @emontnemery , Thanks for the heads up.
I added `message_thread_id` to event log (since it might actually be useful in automation or for debug) and added the corresponding test in tests.